google/resumable_media/requests/download.py

```diff
@@ -194,44 +224,64 @@ def _write_to_stream(self, response):
             ~google.resumable_media.common.DataCorruption: If the download's
                 checksum doesn't agree with server-computed checksum.
         """
-        expected_md5_hash = _get_expected_md5(
-            response, self._get_headers, self.media_url
-        )
 
-        if expected_md5_hash is None:
-            md5_hash = _DoNothingHash()
+        if checksum not in ["md5", "crc32c", None]:
+            raise ValueError("checksum must be ``'md5'``, ``'crc32c'`` or ``None``")
+
+        if checksum:
+            # `_get_expected_checksum()` may return None, in which case it will
+            # emit the info log `_MISSING_CHECKSUM`.
+            expected_checksum = _get_expected_checksum(
+                response, self._get_headers, self.media_url, checksum_type=checksum
+            )
+        else:
+            expected_checksum = None
+
+        if expected_checksum is not None and checksum == "md5":
+            checksum_object = hashlib.md5()
+        elif expected_checksum is not None and checksum == "crc32c":
+            checksum_object = _helpers._get_crc32c_object()
         else:
-            md5_hash = hashlib.md5()
+            checksum_object = _DoNothingHash()
+
         with response:
-            # NOTE: This might "donate" ``md5_hash`` to the decoder and replace
-            #       it with a ``_DoNothingHash``.
             body_iter = response.raw.stream(
                 _helpers._SINGLE_GET_CHUNK_SIZE, decode_content=False
             )
             for chunk in body_iter:
                 self._stream.write(chunk)
-                md5_hash.update(chunk)
+                checksum_object.update(chunk)
             response._content_consumed = True
 
-        if expected_md5_hash is None:
+        if expected_checksum is None:
             return
-
-        actual_md5_hash = base64.b64encode(md5_hash.digest())
-        # NOTE: ``b64encode`` returns ``bytes``, but ``expected_md5_hash``
-        #       came from a header, so it will be ``str``.
-        actual_md5_hash = actual_md5_hash.decode(u"utf-8")
-        if actual_md5_hash != expected_md5_hash:
-            msg = _CHECKSUM_MISMATCH.format(
-                self.media_url, expected_md5_hash, actual_md5_hash
-            )
-            raise common.DataCorruption(response, msg)
-
-    def consume(self, transport):
+        else:
+            actual_checksum = base64.b64encode(checksum_object.digest())
+            # NOTE: ``b64encode`` returns ``bytes``, but ``expected_checksum``
+            #       came from a header, so it will be ``str``.
+            actual_checksum = actual_checksum.decode(u"utf-8")
+            if actual_checksum != expected_checksum:
+                msg = _CHECKSUM_MISMATCH.format(
+                    self.media_url,
+                    expected_checksum,
+                    actual_checksum,
+                    checksum_type=checksum.upper(),
+                )
+                raise common.DataCorruption(response, msg)
```

Comment:
Do you know if data will still be successfully downloaded then raise this DataCorruption() error?

Reply:
I haven't changed this behavior from the previous iteration but my understanding is that actually deleting a download is the responsibility of the user, for instance because we support writing to streams or other situations where it could be unsafe to delete or rewind without user intervention.